src/modelfree/reward_shaping.py

```diff
@@ -0,0 +1,241 @@
+from abc import ABC, abstractmethod
+import numpy as np
+import logging
+import time
+from baselines.common.vec_env import VecEnvWrapper
+from baselines.bench.monitor import ResultsWriter
+from modelfree.simulation_utils import ResettableAgent
+from gym.envs.mujoco import HumanoidEnv
+
+
+class RewardShapingEnv(VecEnvWrapper):
+    """A more direct interface for shaping the reward of the attacking agent."""
+
+    default_shaping_params = {
+        # 'center_reward' : 1,
+        # 'ctrl_cost'     : -1,
+        # 'contact_cost'  : -1,
+        'survive': 10,
+
+        # sparse reward field as per gym_compete/new_envs/multi_agent_env:151
+        'reward_remaining': 1,
+        # dense reward field as per gym_compete/new_envs/agents/humanoid_fighter:45
+        'reward_move': 10
```

Comment:
Suggest leaving trailing commas at end of dicts/lists see https://stackoverflow.com/questions/11597901/why-are-trailing-commas-allowed-in-a-list